### PR TITLE
[Qt6] [Tests] Fix PyQgsJsonUtils and PyQgsOGRProvider

### PR DESCRIPTION
**Pull request by rouault**

### CODE REVIEW

File: tests/src/python/test_provider_ogr.py [hunk 1]
```diff
@@ -1404,7 +1409,16 @@ def testHTTPRequestsOverrider(self):
             # Asked when ogr provider try to open. See QgsOgrProvider::QgsOgrProvider#453 open( OpenModeForceReadOnly );
             handler.add('GET', '/collections/foo', 200, {'Content-Type': 'application/json'}, '{ "id": "foo" }')
 
-            if int(gdal.VersionInfo('VERSION_NUM')) < GDAL_COMPUTE_VERSION(3, 9, 0):
+            # 3.8.3 not necessarily the minimu version
```
Comment (nyalldawson) [inline tests/src/python/test_provider_ogr.py:1412]:
```suggestion
            # 3.8.3 not necessarily the minimum version
```

File: tests/src/python/test_provider_spatialite.py [hunk 2]
```diff
@@ -1561,8 +1561,10 @@ def testSpatialiteDefaultValues(self):
 
         # Test default values
         dp = vl.dataProvider()
-        # FIXME: should it be None?
-        self.assertTrue(dp.defaultValue(0).isNull())
+        if int(QT_VERSION_STR.split('.')[0]) >= 6:
```
Comment (nyalldawson) [inline tests/src/python/test_provider_spatialite.py:1564]: I think this test is actually wrong -- it should be

     self.assertEqual(dp.defaultValue(0), NULL)

That should work in both qt 5 and 6.

File: tests/src/python/test_provider_ogr.py (hunk 1 above)
Comment (rouault) [reply tests/src/python/test_provider_ogr.py]: done

File: tests/src/python/test_provider_spatialite.py (hunk 2 above)
Comment (rouault) [reply tests/src/python/test_provider_spatialite.py]: confirmed, done

Comment (nyalldawson) [reply tests/src/python/test_provider_spatialite.py]: @rouault looks like there's a real bug here -- https://github.com/qgis/QGIS/pull/56578

**Comment by rouault** [comment]
Rebased on top of latest master. I've dropped the "re-enable test_core_ogcutils and PyQgsProviderRegistry" part as they fail on CI for some reason while passing locally for me

File: tests/src/python/test_provider_ogr.py [hunk 3]
```diff
@@ -1092,7 +1092,10 @@ def testBoolFieldEvaluation(self):
         self.assertEqual(vl.fields().at(0).name(), 'bool')
         self.assertEqual(vl.fields().at(0).type(), QVariant.Bool)
         self.assertEqual([f[0] for f in vl.getFeatures()], [True, False, NULL])
-        self.assertEqual([f[0].__class__.__name__ for f in vl.getFeatures()], ['bool', 'bool', 'QVariant'])
```
Comment (nyalldawson) [inline tests/src/python/test_provider_ogr.py:1095]: Do you think we could safely drop this test? I think it's already covered anyway by line 1094

Comment (rouault) [reply tests/src/python/test_provider_ogr.py:1095]: agreed. I've just dropped it

**Review by nyalldawson — APPROVED** [review]

**Pull request merged**